src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -226,93 +171,117 @@ export default function TimeSeriesChart(props: TimeSeriesChartProps) {
         dataToDisplay = props.data;
     }
 
+    let yAxisDomain: AxisDomain | undefined = undefined;
+    if (props.options && props.chartType === "Line") {
+        let domainMin = (props.options as MultiSeriesLineChartOptions)?.domainMin;
+        if (domainMin === "Auto") {
+            yAxisDomain = ["auto", "auto"];
+        } else if (domainMin !== undefined) {
+            yAxisDomain = [domainMin, "auto"];
+        }
+    }
+
+    const xAxisTicks = getXAxisTicks();
+
     return <div className="mdhui-time-series-chart" ref={props.innerRef}>
         {props.title &&
             <CardTitle title={props.title}></CardTitle>
         }
         <div className="chart-container">
-            {(!props.chartHasData) &&
-                <div>
-                    {!!dataToDisplay &&
-                        <div className="no-data-label">No Data</div>
+            {!props.chartHasData && !!dataToDisplay && <div className="no-data-label">No Data</div>}
+            {!props.chartHasData && !dataToDisplay && <LoadingIndicator />}
+            <ResponsiveContainer width="100%" height={150}>
+                <ComposedChart data={dataToDisplay} syncId={props.syncId}>
```

Comment:
The `ComposedChart` allows us to overlay different types of chart components (i.e. Lines with Bars, etc).  This is the main functional change in this PR.

Once all of the charts were being created as  `ComposedChart` components, there was no need to repeat the boilerplate container and chart tags, so now those are at the higher level and only the conditional chart-specific stuff is rendered within the conditionals (per chart type) below.

This also meant the standard chart components were only going to be used in one place, so there was no longer a need to have that extracted to a helper function.  Those components are just rendered inline now.

Reply:
The ComposedChart supports an accessibilityLayer property. Any reason why we wouldn't want to add that now? Info in the [storybook ](https://recharts.org/en-US/storybook)

Reply:
I'm not familiar with the `accessibilityLayer` property and I'm unable to find it in any of the API docs.  The only place I see it mentioned is in that storybook, but I'm not sure what it is supposed to be doing.

Do you have any further info you can link me to or can you describe what it would do for us?